src/ops/validate.rs

```diff
@@ -146,18 +146,18 @@ impl ValidateOp for super::Conditional {
         children: impl DoubleEndedIterator<Item = (NodeIndex, &'a OpType)>,
     ) -> Result<(), ChildrenValidationError> {
         let children = children.collect_vec();
-        // The first input to the ɣ-node is a predicate of Sum type,
+        // The first input to the ɣ-node is a value of Sum type,
         // whose arity matches the number of children of the ɣ-node.
-        if self.predicate_inputs.len() != children.len() {
-            return Err(ChildrenValidationError::InvalidConditionalPredicate {
+        if self.choice_inputs.len() != children.len() {
+            return Err(ChildrenValidationError::InvalidConditionalChoice {
                 child: children[0].0, // Pass an arbitrary child
                 expected_count: children.len(),
-                actual_count: self.predicate_inputs.len(),
-                actual_predicate_rows: self.predicate_inputs.clone(),
+                actual_count: self.choice_inputs.len(),
+                actual_choice_rows: self.choice_inputs.clone(),
```

Comment:
just a side note, elsewhere this would be "choice_variants". I wonder whether "choice_rows" would be better everywhere

Reply:
yes I've tried to go with "tuple_sum_rows" everywhere